src/ui/navigation/AppNavigator.tsx

```diff
@@ -250,7 +250,7 @@ function AppNavigator() {
     <AppStack.Navigator screenOptions={overlayScreenOptions}>
       {isPnPromptNeeded ? (
         <AppStack.Screen name={routeKeys.permissionGrantingPromptScreen}>
-          {() => <PermissionGrantingPrompt skipPnPermission={skipPnPermission} />}
+          {() => <PermissionGrantingPrompt skipPnPermission={skipPnPermission} isSkip={false} />}
```

Comment:
You can make the `isSkip` optional in the component and pass it only when necessary.